keboola/provider.go

```diff
@@ -0,0 +1,166 @@
+package keboola
+
+import (
+	"context"
+	"os"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/provider"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/keboola/go-client/pkg/client"
+	"github.com/keboola/go-client/pkg/storageapi"
+)
+
+// Ensure the implementation satisfies the expected interfaces
+var (
+	_ provider.Provider = &keboolaProvider{}
+)
+
+const KBC_HOST = "KBC_HOST"
+const KBC_TOKEN = "KBC_TOKEN"
+
+// New is a helper function to simplify provider server and testing implementation.
+func New() provider.Provider {
+	return &keboolaProvider{}
+}
+
+// keboolaProvider is the provider implementation.
+type keboolaProvider struct{}
+
+// keboolaProviderModel maps provider schema data to a Go type.
+type keboolaProviderModel struct {
+	Host  types.String `tfsdk:"host"`
+	Token types.String `tfsdk:"token"`
+}
+
+// Metadata returns the provider type name.
+func (p *keboolaProvider) Metadata(_ context.Context, _ provider.MetadataRequest, resp *provider.MetadataResponse) {
+	resp.TypeName = "keboola"
+}
+
+// GetSchema defines the provider-level schema for configuration data.
+func (p *keboolaProvider) GetSchema(_ context.Context) (tfsdk.Schema, diag.Diagnostics) {
+	return tfsdk.Schema{
+		Description: "Interact with Keboola Storage API (https://keboola.docs.apiary.io/).",
+		Attributes: map[string]tfsdk.Attribute{
+			"host": {
+				Description: "URI for Keboola Storage API. May also be provided via " + KBC_HOST + " environment variable.",
+				Type:        types.StringType,
+				Optional:    true,
+			},
+			"token": {
+				Description: "Storage API Token for the Keboola Storage API. May also be provided via " + KBC_TOKEN + " environment variable.",
+				Type:        types.StringType,
+				Sensitive:   true,
+				Optional:    true,
```

Comment:
host aj token su definovane ako optional aby sa mohli nacitat cez envs. V kazdom pripade ak su aj envs prazdne aj tak to spravne spadne chybou.